LoadAsRoot Injector

Beta16 changed Injectors a little bit:
- Injector.getOptional() was changed into Injector.get(token, notFoundValue) to make implementing injectors simpler

We use loadAsRoot in our project, so i took a look at the documentation to see if anything changed:

- You can optionally provide injector and this Injector will be used to instantiate the Component.

I thought there was no problem but then i started getting this error:

```
EXCEPTION: Error: Uncaught (in promise): TypeError: Cannot read property 'get' of null
browser_adapter.ts:78 EXCEPTION: Error: Uncaught (in promise): TypeError: Cannot read property 'get' of nullBrowserDomAdapter.logError @ browser_adapter.ts:78BrowserDomAdapter.logGroup @ browser_adapter.ts:89ExceptionHandler.call @ exception_handler.ts:53(anonymous function) @ application_ref.ts:304schedulerFn @ async.ts:131SafeSubscriber.__tryOrUnsub @ Subscriber.ts:240SafeSubscriber.next @ Subscriber.ts:192Subscriber._next @ Subscriber.ts:133Subscriber.next @ Subscriber.ts:93Subject._finalNext @ Subject.ts:154Subject._next @ Subject.ts:144Subject.next @ Subject.ts:90EventEmitter.emit @ async.ts:117NgZone._zoneImpl.ng_zone_impl_1.NgZoneImpl.onError @ ng_zone.ts:138NgZoneImpl.inner.inner.fork.onHandleError @ ng_zone_impl.ts:90ZoneDelegate.handleError @ zone.js:327Zone.runGuarded @ zone.js:233_loop_1 @ zone.js:487drainMicroTaskQueue @ zone.js:494ZoneTask.invoke @ zone.js:426
browser_adapter.ts:78 STACKTRACE:BrowserDomAdapter.logError @ browser_adapter.ts:78ExceptionHandler.call @ exception_handler.ts:56(anonymous function) @ application_ref.ts:304schedulerFn @ async.ts:131SafeSubscriber.__tryOrUnsub @ Subscriber.ts:240SafeSubscriber.next @ Subscriber.ts:192Subscriber._next @ Subscriber.ts:133Subscriber.next @ Subscriber.ts:93Subject._finalNext @ Subject.ts:154Subject._next @ Subject.ts:144Subject.next @ Subject.ts:90EventEmitter.emit @ async.ts:117NgZone._zoneImpl.ng_zone_impl_1.NgZoneImpl.onError @ ng_zone.ts:138NgZoneImpl.inner.inner.fork.onHandleError @ ng_zone_impl.ts:90ZoneDelegate.handleError @ zone.js:327Zone.runGuarded @ zone.js:233_loop_1 @ zone.js:487drainMicroTaskQueue @ zone.js:494ZoneTask.invoke @ zone.js:426
browser_adapter.ts:78 Error: Uncaught (in promise): TypeError: Cannot read property 'get' of null
    at resolvePromise (zone.js:538)
    at resolvePromise (zone.js:523)
    at zone.js:571
    at ZoneDelegate.invokeTask (zone.js:356)
    at Object.NgZoneImpl.inner.inner.fork.onInvokeTask (ng_zone_impl.ts:56)
    at ZoneDelegate.invokeTask (zone.js:355)
    at Zone.runTask (zone.js:256)
    at drainMicroTaskQueue (zone.js:474)
    at XMLHttpRequest.ZoneTask.invoke (zone.js:426)BrowserDomAdapter.logError @ browser_adapter.ts:78ExceptionHandler.call @ exception_handler.ts:57(anonymous function) @ application_ref.ts:304schedulerFn @ async.ts:131SafeSubscriber.__tryOrUnsub @ Subscriber.ts:240SafeSubscriber.next @ Subscriber.ts:192Subscriber._next @ Subscriber.ts:133Subscriber.next @ Subscriber.ts:93Subject._finalNext @ Subject.ts:154Subject._next @ Subject.ts:144Subject.next @ Subject.ts:90EventEmitter.emit @ async.ts:117NgZone._zoneImpl.ng_zone_impl_1.NgZoneImpl.onError @ ng_zone.ts:138NgZoneImpl.inner.inner.fork.onHandleError @ ng_zone_impl.ts:90ZoneDelegate.handleError @ zone.js:327Zone.runGuarded @ zone.js:233_loop_1 @ zone.js:487drainMicroTaskQueue @ zone.js:494ZoneTask.invoke @ zone.js:426
zone.js:461 Unhandled Promise rejection: Cannot read property 'get' of null ; Zone: angular ; Task: Promise.then ; Value: TypeError: Cannot read property 'get' of null(…)consoleError @ zone.js:461_loop_1 @ zone.js:490drainMicroTaskQueue @ zone.js:494ZoneTask.invoke @ zone.js:426
zone.js:463 Error: Uncaught (in promise): TypeError: Cannot read property 'get' of null(…)
```
Here's the code, working until Beta15:

```typescript
 return System.import(path).then((element) => {
                  return this.dynamicComponentLoader
                      .loadAsRoot(element[type],
                      "#test", null)
                      .then((ref) => {
                          console.log("Component loaded!");
                      })
         });
```
We noticed that we can fix it if we explicitly declare the injector, like this:

```typescript
 return System.import(path).then((element) => {
                  return this.dynamicComponentLoader
                      .loadAsRoot(element[type],
                      "#test", this.injector)
                      .then((ref) => {
                          console.log("Component loaded!");
                      })
         });
```

Why we get this error if providing injectors is optional, according to the documentation?
(even if it doesn't seem to be optional, looking ath the method)

```typescript
    abstract loadAsRoot(type: Type, overrideSelectorOrNode: string | any, injector: Injector, onDispose?: () => void, projectableNodes?: any[][]): Promise<ComponentRef<any>>;
```

--- a/modules/@angular/core/src/linker/dynamic_component_loader.ts
+++ b/modules/@angular/core/src/linker/dynamic_component_loader.ts
@@ -21,8 +21,7 @@ export abstract class DynamicComponentLoader {
    *
    * If needed, the component's selector can be overridden via `overrideSelector`.
    *
-   * You can optionally provide `injector` and this {@link Injector} will be used to instantiate the
-   * Component.
+   * A provided {@link Injector} will be used to instantiate the Component.
    *
    * To be notified when this Component instance is destroyed, you can also optionally provide
    * `onDispose` callback.

